File: src/Shapes.hs
module Shapes where
import qualified Internal.Shapes
import           Types (
    Color
  , Rectangle
  , Vector2
  )

drawPixel :: Int -> Int -> Color -> IO ()
drawPixel posX posY color = Internal.Shapes.drawPixel posX posY color

drawPixelV :: Vector2 -> Color -> IO ()
drawPixelV position color = Internal.Shapes.drawPixelV position color

drawLine :: Int -> Int -> Int -> Int -> Color -> IO ()
drawLine startPosX startPosY endPosX endPosY color = Internal.Shapes.drawLine startPosX startPosY endPosX endPosY color

drawLineV :: Vector2 -> Vector2 -> Color -> IO ()
drawLineV startPos endPos color = Internal.Shapes.drawLineV startPos endPos color

drawLineEx :: Vector2 -> Vector2 -> Float -> Color -> IO ()
drawLineEx startPos endPos thick color = Internal.Shapes.drawLineEx startPos endPos thick color

drawLineBezier :: Vector2 -> Vector2 -> Float -> Color -> IO ()
drawLineBezier startPos endPos thick color = Internal.Shapes.drawLineBezier startPos endPos thick color

drawCircle :: Int -> Int -> Float -> Color -> IO ()
drawCircle centerX centerY radius color = Internal.Shapes.drawCircle centerX centerY radius color

drawCircleGradient :: Int -> Int -> Float -> Color -> Color -> IO ()
drawCircleGradient centerX centerY radius color1 color2 = Internal.Shapes.drawCircleGradient centerX centerY radius color1 color2

drawCircleV :: Vector2 -> Float -> Color -> IO ()
drawCircleV center radius color = Internal.Shapes.drawCircleV center radius color

drawCircleLines :: Int -> Int -> Float -> Color -> IO ()
drawCircleLines centerX centerY radius color = Internal.Shapes.drawCircleLines centerX centerY radius color

drawRectangle :: Int -> Int -> Int -> Int -> Color -> IO ()
drawRectangle posX posY width height color = Internal.Shapes.drawRectangle posX posY width height color

drawRectangleV :: Vector2 -> Vector2 -> Color -> IO ()
drawRectangleV position size color = Internal.Shapes.drawRectangleV position size color

drawRectangleRec :: Rectangle -> Color -> IO ()
drawRectangleRec rec color = Internal.Shapes.drawRectangleRec rec color

drawRectanglePro :: Rectangle -> Vector2 -> Float -> Color -> IO ()
drawRectanglePro rec origin rotation color = Internal.Shapes.drawRectanglePro rec origin rotation color

drawRectangleGradientV :: Int -> Int -> Int -> Int -> Color -> Color -> IO ()
drawRectangleGradientV posX posY width height color1 color2 = Internal.Shapes.drawRectangleGradientV posX posY width height color1 color2

drawRectangleGradientH :: Int -> Int -> Int -> Int -> Color -> Color -> IO ()
drawRectangleGradientH posX posY width height color1 color2 = Internal.Shapes.drawRectangleGradientH posX posY width height color1 color2

drawRectangleGradientEx :: Rectangle -> Color -> Color -> Color -> Color -> IO ()
drawRectangleGradientEx rec col1 col2 col3 col4 = Internal.Shapes.drawRectangleGradientEx rec col1 col2 col3 col4

drawRectangleLines :: Int -> Int -> Int -> Int -> Color -> IO ()
drawRectangleLines posX posY width height color = Internal.Shapes.drawRectangleLines posX posY width height color

drawRectangleLinesEx :: Rectangle -> Int -> Color -> IO ()
drawRectangleLinesEx rec lineThick color = Internal.Shapes.drawRectangleLinesEx rec lineThick color

drawTriangle :: Vector2 -> Vector2 -> Vector2 -> Color -> IO ()
drawTriangle v1 v2 v3 color = Internal.Shapes.drawTriangle v1 v2 v3 color

drawTriangleLines :: Vector2 -> Vector2 -> Vector2 -> Color -> IO ()
drawTriangleLines v1 v2 v3 color = Internal.Shapes.drawTriangleLines v1 v2 v3 color

drawPoly :: Vector2 -> Int -> Float -> Float -> Color -> IO ()
drawPoly center sides radius rotation color = Internal.Shapes.drawPoly center sides radius rotation color

drawPolyEx :: [Vector2] -> Color -> IO ()
drawPolyEx = _todo

drawPolyExLines :: [Vector2] -> Color -> IO ()
drawPolyExLines = _todo

File: cbits/shapes.c
#include <stdlib.h>
#include "raylib.h"

void WrappedDrawPixel(int posX, int posY, Color *color) {
  DrawPixel(posX, posY, *color);
}

void WrappedDrawPixelV(Vector2 *position, Color *color) {
  DrawPixelV(*position, *color);
}

void WrappedDrawLine(int startPosX, int startPosY, int endPosX, int endPosY, Color *color) {
  DrawLine(startPosX, startPosY, endPosX, endPosY, *color);
}

void WrappedDrawLineV(Vector2 *startPos, Vector2 *endPos, Color *color) {
  DrawLineV(*startPos, *endPos, *color);
}

void WrappedDrawLineEx(Vector2 *startPos, Vector2 *endPos, float thick, Color *color) {
  DrawLineEx(*startPos, *endPos, thick, *color);
}

void WrappedDrawLineBezier(Vector2 *startPos, Vector2 *endPos, float thick, Color *color) {
  DrawLineBezier(*startPos, *endPos, thick, *color);
}

void WrappedDrawCircle(int centerX, int centerY, float radius, Color *color) {
  DrawCircle(centerX, centerY, radius, *color);
}

void WrappedDrawCircleGradient(int centerX, int centerY, float radius, Color *color1, Color *color2) {
  DrawCircleGradient(centerX, centerY, radius, *color1, *color2);
}

void WrappedDrawCircleV(Vector2 *center, float radius, Color *color) {
  DrawCircleV(*center, radius, *color);
}

void WrappedDrawCircleLines(int centerX, int centerY, float radius, Color *color) {
  DrawCircleLines(centerX, centerY, radius, *color);
}

void WrappedDrawRectangle(int posX, int posY, int width, int height, Color *color) {
  DrawRectangle(posX, posY, width, height, *color);
}

void WrappedDrawRectangleV(Vector2 *position, Vector2 *size, Color *color) {
  DrawRectangleV(*position, *size, *color);
}

void WrappedDrawRectangleRec(Rectangle *rec, Color *color) {
  DrawRectangleRec(*rec, *color);
}

void WrappedDrawRectanglePro(Rectangle *rec, Vector2 *origin, float rotation, Color *color) {
  DrawRectanglePro(*rec, *origin, rotation, *color);
}

void WrappedDrawRectangleGradientV(int posX, int posY, int width, int height, Color *color1, Color *color2) {
  DrawRectangleGradientV(posX, posY, width, height, *color1, *color2);
}

void WrappedDrawRectangleGradientH(int posX, int posY, int width, int height, Color *color1, Color *color2) {
  DrawRectangleGradientH(posX, posY, width, height, *color1, *color2);
}

void WrappedDrawRectangleGradientEx(Rectangle *rec, Color *col1, Color *col2, Color *col3, Color *col4) {
  DrawRectangleGradientEx(*rec, *col1, *col2, *col3, *col4);
}

void WrappedDrawRectangleLines(int posX, int posY, int width, int height, Color *color) {
  DrawRectangleLines(posX, posY, width, height, *color);
}

void WrappedDrawRectangleLinesEx(Rectangle *rec, int lineThick, Color *color) {
  DrawRectangleLinesEx(*rec, lineThick, *color);
}

void WrappedDrawTriangle(Vector2 *v1, Vector2 *v2, Vector2 *v3, Color *color) {
  DrawTriangle(*v1, *v2, *v3, *color);
}

void WrappedDrawTriangleLines(Vector2 *v1, Vector2 *v2, Vector2 *v3, Color *color) {
  DrawTriangleLines(*v1, *v2, *v3, *color);
}

void WrappedDrawPoly(Vector2 *center, int sides, float radius, float rotation, Color *color) {
  DrawPoly(*center, sides, radius, rotation, *color);
}

void WrappedDrawPolyEx(Vector2 *points, int numPoints, Color *color) {
  DrawPolyEx(points, numPoints, *color);
}

void WrappedDrawPolyExLines(Vector2 *points, int numPoints, Color *color) {
  DrawPolyExLines(points, numPoints, *color);
}
